internal/clients/http/azure/create_vm.go

```diff
@@ -453,7 +454,7 @@ func (c *client) prepareVirtualMachineParameters(location string, instanceType a
 						PublicKeys: []*armcompute.SSHPublicKey{
 							{
 								Path:    to.Ptr(fmt.Sprintf("/home/%s/.ssh/authorized_keys", adminUsername)),
-								KeyData: to.Ptr(sshKeyBody),
+								KeyData: to.Ptr(vmParams.Pubkey.Body),
```

Comment:
Can Pubkey be 'nil' here in some cases? :thinking: 
Since this is a pointer, we do not have to makeup values, so we could in theory do:
```
keydata := nil
if vmParams.Pubkey != nil {
    keydata = to.Ptr(vmParams.Pubkey.Body)
}
...
KeyData: keydata,
```
This violates the happy code path we want to keep, but this is just an idea.

Reply:
You are right, pubkey does not need to be pointer in the vmParams type, I can remove that. We fetch it from the database and in case it does not exist it is an error, so it is guaranteed to be present.